### PR TITLE
Fix analysis step to use only a single helper-container-host executor

### PR DESCRIPTION
**Pull request by bensze01**
This PR restores the intended behaviour of the `analysis` step, that it runs on a `helper-container-host` instance.

The current upstream code reserves a `helper-container-host` but actually runs all the tests on a dfferent, `container-host` executor

Test runs:
- OpenCI
  - Supposed to pass
    - development: [PR-head][openci-development-head] [PR-merge][openci-development-merge]
    - mbedtls-3.6: [PR-head][openci-3.6-head] [PR-merge][openci-3.6-merge]
    - mbedtls-2.28: [PR-head][openci-2.28-head] [PR-merge][openci-2.28-merge]
  - Supposed to fail
    - [Fail all CMake builds][openci-cmake]
    - [Unexecuted test][openci-unexecuted]
 - Internal CI
  - Supposed to pass
    - development: [PR-head][internal-development-head] [PR-merge][internal-development-merge]
    - mbedtls-3.6: [PR-head][internal-3.6-head] [PR-merge][internal-3.6-merge]
    - mbedtls-2.28: [PR-head][internal-2.28-head] [PR-merge][internal-2.28-merge]
  - Supposed to fail
    - [Fail all CMake builds][internal-cmake]
    - [Unexecuted test][internal-unexecuted]
    
[openci-development-head]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-906-head/23/
[openci-development-merge]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-906-merge/24/
[openci-3.6-head]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-1233-head/3/
[openci-3.6-merge]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-1233-merge/6/
[openci-2.28-head]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-850-head/12/
[openci-2.28-merge]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-850-merge/10/
[openci-cmake]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-759-head/12/
[openci-unexecuted]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-919-head/3/
[internal-development-head]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-906-head/122/
[internal-development-merge]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-906-merge/157/
[internal-3.6-head]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-1233-head/8/
[internal-3.6-merge]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-1233-merge/8/
[internal-2.28-head]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-850-head/10/
[internal-2.28-merge]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-850-merge/10/
[internal-cmake]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-759-head/9/
[internal-unexecuted]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-919-head/1/

### CODE REVIEW

File: vars/analysis.groovy [hunk 1]
```diff
@@ -285,14 +288,5 @@ def analyze_results(BranchInfo info) {
     if (outcome_stashes.isEmpty()) {
         return
     }
-    node_record_timestamps('helper-container-host', 'result-analysis') {
-        dir('outcomes') {
-            deleteDir()
```
Comment (paul-elliott-arm) [inline vars/analysis.groovy:290]: I'm not sure I understand this piece of code at all (it deletes the directory before and after `process_outcomes()`, but I thought I had better check we no longer needed it, as its not referenced in the commit message.

**Review by paul-elliott-arm — COMMENTED** [review]
One question, otherwise looks good, dependant on test runs

File: vars/analysis.groovy [hunk 2]
```diff
@@ -227,10 +227,17 @@ def stash_outcomes(job_name) {
...
-// from all the jobs.
-def process_outcomes(BranchInfo info) {
-    String job_name = 'outcome_analysis'
+/** Process the outcome files from all the jobs */
+def analyze_results(BranchInfo info) {
+    // After running on an old branch which doesn't have the outcome
+    // file generation mechanism, or after running a partial run,
+    // there may not be any outcome file. In this case, silently
+    // do nothing.
+    if (outcome_stashes.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (gilles-peskine-arm) [reply vars/analysis.groovy:236]: This is preexisting, dating back from the very first PR that introduced outcome analysis.

We'll silently skip running outcome analysis if there are no outcome files. Can you think of any other “catastrophic problems”?

Anyway, considerations for branches that don't have outcome file generation have run their course. I think we still should avoid an uncaught exception if there are no outcome files at all (I'm not sure if that's still a risk with the current structure of the code). But we should probably only skip outcome analysis if there are no outcome files _and there has already been a failure_.

It's out of scope here though.

File: vars/analysis.groovy (hunk 1 above)
Comment (bensze01) [reply vars/analysis.groovy:290]: `gen_docker_job()` also calls `deleteDir()` internally [here][1] and [here][2], which is functionally equivalent to these calls.

[1]: https://github.com/Mbed-TLS/mbedtls-test/blob/12d87533661ddcea675052d7b48ab0a07c754c49/vars/gen_jobs.groovy#L135
[2]: https://github.com/Mbed-TLS/mbedtls-test/blob/12d87533661ddcea675052d7b48ab0a07c754c49/vars/gen_jobs.groovy#L181

**Review by paul-elliott-arm — APPROVED** [review]
LGTM

**Review by tom-cosgrove-arm — APPROVED** [review]

**Pull request merged**